build.sbt

```diff
@@ -200,236 +231,92 @@ lazy val circe = project.in(file("."))
   .settings(
     libraryDependencies ++= Seq(
       "org.scalacheck" %%% "scalacheck" % scalaCheckVersion % "test",
       "org.scalatest" %%% "scalatest" % scalaTestVersion % "test"
     )
   )
-  .jvmSettings(
-    mimaPreviousArtifacts := Set("io.circe" %% "circe-numbers" % previousCirceVersion)
-  )
-  .jvmConfigure(_.copy(id = "numbers"))
-  .jsConfigure(_.copy(id = "numbersJS"))
 
 lazy val numbers = numbersBase.jvm
 lazy val numbersJS = numbersBase.js
 
-lazy val coreBase = crossProject.in(file("modules/core"))
-  .settings(
-    description := "circe core",
-    moduleName := "circe-core",
-    name := "Circe core",
-    crossScalaVersions := scalaVersions
-  )
-  .settings(allSettings: _*)
+lazy val coreBase = circeCrossModule("core", mima = previousCirceVersion)
   .settings(
-    libraryDependencies ++= Seq(
-      "org.typelevel" %%% "cats-core" % catsVersion
-    ),
+    libraryDependencies += "org.typelevel" %%% "cats-core" % catsVersion,
     sourceGenerators in Compile += (sourceManaged in Compile).map(Boilerplate.gen).taskValue
   )
-  .jvmSettings(
-    mimaPreviousArtifacts := Set("io.circe" %% "circe-core" % previousCirceVersion)
-  )
-  .jvmConfigure(_.copy(id = "core"))
-  .jsConfigure(_.copy(id = "coreJS"))
   .dependsOn(numbersBase)
 
 lazy val core = coreBase.jvm
 lazy val coreJS = coreBase.js
 
-lazy val genericBase = crossProject.in(file("modules/generic"))
-  .settings(
-    description := "circe generic",
-    moduleName := "circe-generic",
-    name := "Circe generic",
-    crossScalaVersions := scalaVersions
-  )
-  .settings(allSettings: _*)
-  .settings(macroDependencies: _*)
+lazy val genericBase = circeCrossModule("generic", mima = previousCirceVersion)
+  .settings(macroDependencies)
   .settings(
-    libraryDependencies += "com.chuusai" %%% "shapeless" % shapelessVersion,
-    sources in (Compile, doc) := (
-      CrossVersion.partialVersion(scalaVersion.value) match {
-        case Some((2, 11)) => (sources in (Compile, doc)).value
-        case _ => Nil
-      }
-    )
-  )
-  .jvmSettings(
-    mimaPreviousArtifacts := Set("io.circe" %% "circe-generic" % previousCirceVersion)
+    libraryDependencies += "com.chuusai" %%% "shapeless" % shapelessVersion
   )
-  .jvmConfigure(_.copy(id = "generic"))
-  .jsConfigure(_.copy(id = "genericJS"))
   .dependsOn(coreBase)
 
 lazy val generic = genericBase.jvm
 lazy val genericJS = genericBase.js
 
-lazy val genericExtrasBase = crossProject.crossType(CrossType.Pure).in(file("modules/generic-extras"))
-  .settings(
-    description := "circe generic extras",
-    moduleName := "circe-generic-extras",
-    name := "Circe generic extras",
-    crossScalaVersions := scalaVersions
-  )
-  .settings(allSettings: _*)
-  .settings(macroDependencies: _*)
-  .settings(
-    sources in (Compile, doc) := (
-      CrossVersion.partialVersion(scalaVersion.value) match {
-        case Some((2, 11)) => (sources in (Compile, doc)).value
```

Comment:
I removed this (also in other projects) because unidoc has its own filter and this was not updated to 2.12 so most likely it is not used.